### ISSUE TITLE
Replace deprecated util._extend API with Object.assign()

### ISSUE THREAD

**Issue by terrencekhng** [Unitech/pm2#5238]
## What's going wrong?
Reference: https://nodejs.org/api/util.html#util_extendtarget-source
The util._extend() method was never intended to be used outside of internal Node.js modules. There are some modules in pm2 still using this method, and it is hereafter suggested to be replaced.

## How could we reproduce this issue?
N/A

## Supporting information
N/A

```
$ pm2 report
```

**Comment by enkr1** [comment]
any update on this?

**Comment by eliliam** [comment]
@Unitech could we get an update since it's been three years since the issue was created? I'm also seeing this warning happening more frequently now.

**Comment by NicmeisteR** [comment]
Yeah having this error pop up consistently is a bit of a pain , also hoping for a fix soon.

**Comment by egoroof** [comment]
Should be fixed in https://github.com/keymetrics/pm2-io-apm/pull/303

**Comment by egoroof** [comment]
should be fixed in [v5.4.2](https://github.com/Unitech/pm2/releases/tag/v5.4.2)

**Comment by Unitech** [comment]
Should be fixed in **v7.0.0** — https://github.com/Unitech/pm2/releases/tag/v7.0.0

To upgrade:

```bash
# With npm
npm install pm2@latest -g

# Or with bun
bun install pm2@latest -g

# Then reload the daemon and restore your process list
pm2 update
```

`pm2 update` dumps the current process list, kills the running daemon, starts the new version, and restores all processes — so existing apps keep running with no manual restart.

Please reopen if the issue still occurs on 7.0.0+.

**Comment by AntonAgaronov** [comment]
The problem occurs in 7.0.1
